File: app/src/main/java/com/numero/sojodia/repository/ConfigRepository.java
package com.numero.sojodia.repository;

import android.content.Context;
import android.content.SharedPreferences;

import com.numero.sojodia.util.DateUtil;

public class ConfigRepository implements IConfigRepository {

    private static final String PREFERENCES = "PREFERENCES";
    private static final String VERSION_CODE = "FIRST_BOOT";
    private static final String DATE = "DATE";

    private long versionCode = -1;

    private final SharedPreferences sharedPreferences;

    public ConfigRepository(Context context) {
        sharedPreferences = context.getSharedPreferences(PREFERENCES, Context.MODE_PRIVATE);
    }

    @Override
    public boolean isTodayUpdateChecked() {
        String previousCheckDateString = sharedPreferences.getString(DATE, "");
        return DateUtil.getTodayStringOnlyFigure().equals(previousCheckDateString);
    }

    @Override
    public void successUpdate() {
        SharedPreferences.Editor edit = sharedPreferences.edit();
        edit.putLong(VERSION_CODE, versionCode);
        edit.apply();
    }

    @Override
    public boolean canUpdate() {
        return sharedPreferences.getLong(VERSION_CODE, 20170401L) < versionCode;
    }

    public void setVersionCode(long versionCode) {
        this.versionCode = versionCode;
        // バージョンがセットされた時はチェックに成功した場合
        SharedPreferences.Editor edit = sharedPreferences.edit();
        edit.putString(DATE, DateUtil.getTodayStringOnlyFigure());
        edit.apply();
    }

    public long getVersionCode() {
        return sharedPreferences.getLong(VERSION_CODE, 20170401L);
    }
}

File: app/src/main/kotlin/com/numero/sojodia/repository/ConfigRepository.kt
package com.numero.sojodia.repository

import android.content.Context
import android.content.SharedPreferences
import androidx.core.content.edit
import com.numero.sojodia.extension.getTodayStringOnlyFigure
import com.numero.sojodia.model.AppTheme
import com.numero.sojodia.model.CurrentVersion
import com.numero.sojodia.model.LatestVersion
import java.util.*

class ConfigRepository(context: Context) : IConfigRepository {

    private val sharedPreferences: SharedPreferences = context.getSharedPreferences(PREFERENCES, Context.MODE_PRIVATE)

    override val currentVersion: CurrentVersion
        get() {
            val version = sharedPreferences.getLong(VERSION_CODE, DEFAULT_DATA_VERSION)
            return CurrentVersion(version)
        }

    override val isTodayUpdateChecked: Boolean
        get() {
            return Calendar.getInstance().getTodayStringOnlyFigure() == sharedPreferences.getString(DATE, "")
        }

    override fun updateBusDataVersion(latestVersion: LatestVersion) {
        sharedPreferences.edit {
            putLong(VERSION_CODE, latestVersion.value)
        }
    }

    override fun updateCheckUpdateDate() {
        sharedPreferences.edit {
            putString(DATE, Calendar.getInstance().getTodayStringOnlyFigure())
        }
    }

    override var appTheme: AppTheme
        set(value) {
            sharedPreferences.edit {
                putString(KEY_APP_THEME, value.key)
            }
        }
        get() {
            val entry = sharedPreferences.getString(KEY_APP_THEME, AppTheme.LIGHT.key)
            return AppTheme.from(entry)
        }

    companion object {

        const val DEFAULT_DATA_VERSION = 20170401L

        private const val PREFERENCES = "PREFERENCES"
        private const val VERSION_CODE = "FIRST_BOOT"
        private const val KEY_APP_THEME = "KEY_APP_THEME"
        private const val DATE = "DATE"
    }
}
